src/components/dex/Charts.vue

```diff
@@ -305,41 +320,43 @@ export default {
               if (!this.tradingView || !this.tradingView._options) {
                 return;
               }
-              const to = parseInt((new Date().valueOf()) / 1000, 10)
-              const from = to - 120
+              const to = Math.round(new Date().valueOf() / 1000);
+              const from = to - (resolutionInMinutes * 60 * 2); //  back 2x resolutions
 
-              this.tradingView._options.datafeed.getBars(_symbolInfo, resolution, from, to, (bars) => {
-                if (bars.length === 0) {
-                  return;
-                }
+              const bars = this.$store.state.tradeHistory && this.$store.state.tradeHistory.getBars ?
+                  this.$store.state.tradeHistory.getBars(this.$store.state.tradeHistory, resolution, from, to) :
```

Comment:
Are we storing methods in the Vuex store? That's a whole lot of code smell to me.

Reply:
I think it sets a reference to the method. And this is the only place where it happens currently; the reason it needs to happen is related to how the trading chart works. If you deep dive and see a better way to do it, feel free to issue a PR to change it.